contracts/extensions/StreamingFeeSplitExtension.sol

```diff
@@ -0,0 +1,206 @@
+
+        uint256 ownerTake = totalFees.preciseMul(_manager(_setToken).ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            _manager(_setToken).transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            _manager(_setToken).transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes StreamingFeeSplitExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes StreamingFeeSplitExtension to the DelegatedManager and StreamingFeeModule to the SetToken
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     * @param _settings             FeeState struct defining fee parameters for StreamingFeeModule initialization
+     */
+    function initializeModuleAndExtension(
+        IDelegatedManager _delegatedManager,
+        IStreamingFeeModule.FeeState memory _settings
+    ) 
+        external 
+    {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(setToken.isPendingModule(address(streamingFeeModule)), "StreamingFeeModule must be pending");
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        bytes memory callData = abi.encodeWithSignature(
+            "initialize(address,(address,uint256,uint256,uint256))", 
+            setToken,
+            _settings);
+        _invokeManager(setToken, address(streamingFeeModule), callData);
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY MANAGER: Remove an existing SetToken and DelegatedManager tracked by the TradeExtension 
+     */
+    function removeExtension() external override {
+        ISetToken setToken = IDelegatedManager(msg.sender).setToken();
+        require(msg.sender == address(_manager(setToken)), "Must be Manager");
+        delete setManagers[setToken];
```

Comment:
should we emit an event here?

```
event ExtensionRemoved(
    address _setToken,
    address _delegatedManager
);
```

Reply:
Yeah probably just in case. Not worried about gas since this shouldn't be called too often. We'll emit something on the manager as well, but this would make the data a bit easier to search if you wanted to understand the state across different Sets